### PR TITLE
Save HTTP response headers under ${SDKMAN}/var/metadata

### PR DESCRIPTION
**Pull request by hgeraldino**
Store the HTTP response headers for each downloaded candidate. 

These headers will be used in future PRs to extract metadata such as: checksums, file type, file name, etc.

### CODE REVIEW

File: src/test/resources/features/flush.feature [hunk 1]
```diff
@@ -42,3 +44,9 @@ Feature: Flush
 		When I enter "sdk flush temp"
 		Then no "res-1.2.0.zip" file is present in temporary storage
 		And I see "1 archive(s) flushed"
+
+	Scenario: Clear out the metadata
+		Given a headers file "grails-1.3.9.headers" in metadata directory
+		When I enter "sdk flush metadata"
+		Then no metadata is cached
+		And I see "1 archive(s) flushed"
```
Comment (marc0der) [inline src/test/resources/features/flush.feature:52]: Should this read something like "Metadata was flushed"?

File: src/test/groovy/sdkman/steps/installation_steps.groovy [hunk 2]
```diff
@@ -96,3 +96,15 @@ And(~'^I have configured "([^"]*)" to "([^"]*)"$') { String configName, String f
 And(~/^the exit code is (\d+)$/) { Integer rc ->
 	assert bash.getStatus() == rc
 }
+
+And(~'^the response headers file is created for candidate "([^"]*)" and version "([^"]*)"$') { String candidate, String version ->
+	def headersFile = "${metadataDir}/${candidate}-${version}.headers" as File
+	if (!headersFile.exists()) println bash.output
```
Comment (marc0der) [inline src/test/groovy/sdkman/steps/installation_steps.groovy:102]: The printing of the bash output probably shouldn't happen here. I think it is already done elsewhere in the bash environment helper.

File: src/test/groovy/sdkman/steps/installation_steps.groovy [hunk 3]
```diff
@@ -96,3 +96,15 @@ And(~'^I have configured "([^"]*)" to "([^"]*)"$') { String configName, String f
...
+	assert headersFile.exists()
+}
+
+And(~'^no response headers are written for candidate "([^"]*)" and version "([^"]*)"$') { String candidate, String version ->
+	def headersFile = "${metadataDir}/${candidate}-${version}.headers" as File
+	if (headersFile.exists()) println bash.output
```
Comment (marc0der) [inline src/test/groovy/sdkman/steps/installation_steps.groovy:108]: Same as above.

**Review by marc0der — COMMENTED** [review]
This is awesome work! I've left some very minor comments to be addressed.

File: src/test/resources/features/flush.feature (hunk 1 above)
Comment (hgeraldino) [reply src/test/resources/features/flush.feature:52]: This option is calling the same function (`__sdkman_cleanup_folder`) as the other option of the flush command (archives, tmp) so the message logged is the same

Not sure if I want to implement a new function

File: src/test/groovy/sdkman/steps/installation_steps.groovy (hunk 2 above)
Comment (hgeraldino) [reply src/test/groovy/sdkman/steps/installation_steps.groovy:102]: removed

File: src/test/groovy/sdkman/steps/installation_steps.groovy (hunk 3 above)
Comment (hgeraldino) [reply src/test/groovy/sdkman/steps/installation_steps.groovy:108]: removed

File: src/main/bash/sdkman-install.sh [hunk 4]
```diff
@@ -120,11 +120,15 @@ function __sdkman_download() {
 	version="$2"
 
 	archives_folder="${SDKMAN_DIR}/archives"
+	metadata_folder="${SDKMAN_DIR}/var/metadata"
+	mkdir -p ${metadata_folder}
```
Comment (marc0der) [inline src/main/bash/sdkman-install.sh:124]: I think this is fine for now, but the creation of this folder can probably be performed by the install and selfupdate hooks in the future.

**Review by marc0der — APPROVED** [review]
LGTM, thanks @hgeraldino! Are you happy with these initial changes @helpermethod?

**Review by helpermethod — APPROVED** [review]
LGTM! Great work!

**Pull request merged**